lib/generators/site_setting_move_to_groups_migration/site_setting_move_to_groups_migration_generator.rb

```diff
@@ -89,17 +90,35 @@ def down
 
   private
 
+  def load_all_settings
+    load_settings(File.join(Rails.root, "config", "site_settings.yml"))
+
+    if GlobalSetting.load_plugins?
+      Dir[File.join(Rails.root, "plugins", "*", "config", "settings.yml")].each do |file|
+        load_settings(file, plugin: file.split("/")[-3])
+      end
+    end
+
+    if Rails.env.test?
+      load_settings(
+        File.join(Rails.root, "spec", "fixtures", "site_settings", "generator_test.yml"),
+      )
+    end
+  end
+
   def validate_setting_name!(name)
-    if !SiteSetting.respond_to?(name)
+    if !self.respond_to?(name)
       say "Site setting with #{name} does not exist"
       raise ArgumentError
     end
   end
 
   def setting_type(name)
-    @site_settings ||= load_settings(File.join(Rails.root, "config", "site_settings.yml"))
-    subgroup = @site_settings.values.find { |row| row.keys.include?(name) }
-    subgroup[name][:enum]
+    if type_supervisor.get_type(name.to_sym) == :enum
+      return type_supervisor.get_enum_class(name.to_sym)&.to_s
```

Comment:
The safe navigation indicates that it's possible to have an enum type setting without an enum class. Is this what we want, or something we should fix in the type supervisor?

Reply:
Good point, the safe nav is not needed :+1: